Refactor download page components and add DownloadContent

NebraskaCoder opened the pull request:
- Simplified Jest configuration by merging object properties.
- Refactored the Download page to utilize a new DownloadContent component for better organization.
- Updated NavItem component to streamline link rendering.
- Introduced new CloudImage and DefaultImage components with improved structure and localization support.
- Enhanced table components with localized headers and cell content.
- Added internationalization support for various UI elements in the download section.

This addresses issue #40

pwtsai commented:
Hi, thanks for your contribution, just a friendly reminder, few commits are duplicated with #131. BTW, we've created a [place](https://github.com/rockylinuxtw) and [website](https://rockylinux.tw/), trying bridging Rocky Linux and Taiwan's local community. Feel free to PR to the repo if you have anything worthy to share with the community : )

pwtsai commented on i18n/zh-TW.toml:
```diff
@@ -59,7 +59,7 @@ other = "zh-tw"
 
   [landing.p2.heading]
-  other = "為什麼要創建 Rocky Linux 專案？"
+  other = "Rocky Linux 專案是什麼？"
```
The original sentence is not using a word-for-word translation, I think we may discuss a regulation for further situations.

wusung replied:
The translation `Rocky Linux 為何而生？`  is better.   I think you can send your PR freely override mine.   I'll send PR to your branch for zh-tw translation later.   Maybe we could chat in Rocky Slack channel.   https://hpcng.slack.com

pwtsai replied:
Yes, of course. It will be grate if we could have collaboration. While now we have a small situation here (no offense), I think your PR should target to the develop branch...

pwtsai commented:
Hi @hbjydev @wusung,  #139 is directly asked to merge to main branch, while according to the [CONTRIBUTING.md](https://github.com/rocky-linux/rockylinux.org/blob/main/CONTRIBUTING.md), PRs is asked to be targeted towards the develop branch first. Since this PR might conflict to previous #131 (which is already merged into develop branch by @ressonix ), any suggestions ?

NebraskaCoder committed:
Refactor download page components and add DownloadContent

- Simplified Jest configuration by merging object properties.
- Refactored the Download page to utilize a new DownloadContent component for better organization.
- Updated NavItem component to streamline link rendering.
- Introduced new CloudImage and DefaultImage components with improved structure and localization support.
- Enhanced table components with localized headers and cell content.
- Added internationalization support for various UI elements in the download section.

NebraskaCoder committed:
Refactor sorting logic in CloudImage table headers for consistency

- Updated sorting condition in VersionHeader, RegionHeader, and ArchitectureHeader components to use strict equality (===) instead of loose equality (==) for improved code reliability.
